tiledb/sm/array/array.cc

```diff
@@ -211,12 +212,32 @@ Status Array::open(
   RETURN_NOT_OK(
       encryption_key_.set_key(encryption_type, encryption_key, key_length));
 
-  timestamp_ =
-      (query_type == QueryType::READ) ? utils::time::timestamp_now_ms() : 0;
   metadata_.clear();
   metadata_loaded_ = false;
   non_empty_domain_computed_ = false;
 
+  bool found = false;
+  uint64_t timestamp_cfg = 0;
+  RETURN_NOT_OK(config_.get<uint64_t>(
+      "sm.array_timestamp_start", &timestamp_cfg, &found));
+  assert(found);
+
+  if (query_type == QueryType::READ) {
+    if (timestamp_cfg == UINT64_MAX) {
+      timestamp_ = utils::time::timestamp_now_ms();
+    } else {
+      timestamp_ = timestamp_cfg;
+    }
+
+  } else {
+    assert(query_type == QueryType::WRITE);
+    if (timestamp_cfg == UINT64_MAX) {
+      timestamp_ = 0;
+    } else {
+      timestamp_ = timestamp_cfg;
+    }
```

Comment:
Timestamps aren't applicable to writes, refactor this so that:
- `timestamp_ ==0` when the query type is writes
- only fetch the config for read queries.

I'm thinking of something like:
```
if (query_type == QueryType::READ) {
  // read cfg
  // set timestamp based on cfg value
} else {
  assert(query_type == QueryType::WRITE);
  timestamp_ = 0;
}
```

Reply:
The current syntax is required to maintain existing behavior